### PR TITLE
Refactoring the analyzer user guide

### PR DESCRIPTION
**Pull request by dkrupp**

### CODE REVIEW

File: docs/analyzer/user_guide.md [hunk 1]
```diff
@@ -956,7 +1007,7 @@ Environment variables
 </details>
 
 
-### _Skip_ file <a name="skip"></a>
+##### _Skip_ file
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1010]:
```suggestion
#### _Skip_ file
```

File: docs/analyzer/user_guide.md [hunk 2]
```diff
@@ -977,7 +1028,7 @@ checked, `+` means that it should be.
  * Path parts should start and end with `*`.
  * To skip everything use the `-*` mark. **Watch out for the order!**
 
-#### Absolute path examples <a name="skip-abs-example"></a>
+####### Absolute path examples
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1031]:
```suggestion
##### Absolute path examples
```

File: docs/analyzer/user_guide.md [hunk 3]
```diff
@@ -989,7 +1040,7 @@ checked, `+` means that it should be.
 In the above example, every file under `/dir` **will be** skipped, except the
 one explicitly specified to **be analyzed** (`/dir/do.check.this.file`).
 
-#### Relative or partial path examples <a name="skip-rel-example"></a>
+####### Relative or partial path examples
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1043]:
```suggestion
##### Relative or partial path examples
```

File: docs/analyzer/user_guide.md [hunk 4]
```diff
@@ -1010,7 +1061,7 @@ Please note that when `-i SKIPFILE` is used along with `--stats` or
 that statistics and ctu-pre-analysis will be created for *all* files in the
 *compilation database*.
 
-### CodeChecker analyzer configuration <a name="analyzer-configuration"></a>
+##### CodeChecker analyzer configuration
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1064]:
```suggestion
#### CodeChecker analyzer configuration
```

File: docs/analyzer/user_guide.md [hunk 5]
```diff
@@ -1099,15 +1150,15 @@ See [Configure Clang Static Analyzer and checkers](checker_and_analyzer_configur
 documentation for a more detailed description how to use the `saargs`,
 `tidyargs` and `z3` arguments.
 
-#### Analyzer and checker config options <a name="analyzer-checker-config-option"></a>
+###### Analyzer and checker config options
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1153]:
```suggestion
##### Analyzer and checker config options
```

File: docs/analyzer/user_guide.md [hunk 6]
```diff
@@ -1099,15 +1150,15 @@ See [Configure Clang Static Analyzer and checkers](checker_and_analyzer_configur
...
 
 CodeChecker's analyzer module currently handles ClangSA and ClangTidy. The main
 purpose of this analyzer module is to hide the differences between the
 interfaces and parameterization of these two tools. Both ClangSA and ClangTidy
 have a set of checkers with fine-tuning config options and the analyzer tools
 themselves can also be configured.
 
-##### Configuration of analyzer tools <a name="analyzer-config-option"></a>
+###### Configuration of analyzer tools
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1161]:
```suggestion
##### Configuration of analyzer tools
```

File: docs/analyzer/user_guide.md [hunk 7]
```diff
@@ -1146,7 +1197,7 @@ CodeChecker analyze build.json \
   -o reports
 ```
 
-##### Configuration of checkers <a href="checker-config-option"></a>
+###### Configuration of checkers
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1200]:
```suggestion
##### Configuration of checkers
```

File: docs/analyzer/user_guide.md [hunk 8]
```diff
@@ -1192,7 +1243,7 @@ The analyzer and checker configuration options can also be inserted in the
 CodeChecker configuration file. See an example
 [above](#analyzer-configuration-file).
 
-#### Forwarding compiler options <a name="forwarding-compiler-options"></a>
+###### Forwarding compiler options
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1246]:
```suggestion
##### Forwarding compiler options
```

File: docs/analyzer/user_guide.md [hunk 9]
```diff
@@ -1212,7 +1263,7 @@ value. Otherwise an error message is logged saying that the variable is not
 set, and in this case an empty string is inserted in the place of the
 placeholder.
 
-##### <a name="clang-static-analyzer"></a> _Clang Static Analyzer_
+###### _Clang Static Analyzer_
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1266]:
```suggestion
##### _Clang Static Analyzer_
```

File: docs/analyzer/user_guide.md [hunk 10]
```diff
@@ -1229,7 +1280,7 @@ options, for example:
 
 (where `MY_LIB` is the path of a library code)
 
-##### _Clang-Tidy_ <a name="clang-tidy"></a>
+#########_Clang-Tidy_
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1283]:
```suggestion
##### _Clang-Tidy_
```

File: docs/analyzer/user_guide.md [hunk 11]
```diff
@@ -1252,7 +1303,7 @@ Example:
 
 (where `MY_LIB` is the path of a library code)
 
-#### Compiler-specific include path and define detection (cross compilation) <a name="include-path"></a>
+###### Compiler-specific include path and define detection (cross compilation)
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1306]:
```suggestion
##### Compiler-specific include path and define detection (cross compilation)
```

File: docs/analyzer/user_guide.md [hunk 12]
```diff
@@ -1292,7 +1343,7 @@ analysis unless `--keep-gcc-include-fixed` or `--keep-gcc-intrin` flag is
 given. For further information see
 [GCC incompatibilities](gcc_incompatibilities.md).
 
-### Toggling checkers <a name="toggling-checkers"></a>
+##### Toggling checkers
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1346]:
```suggestion
#### Toggling checkers
```

File: docs/analyzer/user_guide.md [hunk 13]
```diff
@@ -1361,7 +1412,7 @@ Checkers are taken into account based on the following order:
 Disabling certain checkers - such as the `core` group - is unsupported by
 the LLVM/Clang community, and thus discouraged.
 
-### Toggling compiler warnings <a name="toggling-warnings"></a>
+##### Toggling compiler warnings
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1415]:
```suggestion
#### Toggling compiler warnings
```

File: docs/analyzer/user_guide.md [hunk 14]
```diff
@@ -1397,7 +1448,7 @@ critical severity bug.
 **Note**: by default `-Wall` and `-Wextra` warnings are enabled.
 
 
-#### Checker profiles <a name="checker-profiles"></a>
+##### Checker profiles
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1451]:
```suggestion
#### Checker profiles
```

File: docs/analyzer/user_guide.md [hunk 15]
```diff
@@ -1415,7 +1466,7 @@ different from checker(-group) names as they are enabled using the same syntax
 and coinciding names could cause unintended behavior.
 
 
-#### `--enable-all` <a name="enable-all"></a>
+####### `--enable-all`
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1469]:
```suggestion
##### `--enable-all`
```

File: docs/analyzer/user_guide.md [hunk 16]
```diff
@@ -1441,7 +1492,7 @@ special checker groups: `alpha.`, `debug.`, `osx.`, `abseil-`, `android-`,
 
 can be used to "further" enable `alpha.` checkers, and disable `misc` ones.
 
-### Cross Translation Unit (CTU) analysis mode <a name="ctu"></a>
+##### Cross Translation Unit (CTU) analysis mode
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1495]:
```suggestion
#### Cross Translation Unit (CTU) analysis mode
```

File: docs/analyzer/user_guide.md [hunk 17]
```diff
@@ -1481,7 +1532,7 @@ cross translation unit analysis arguments:
                         analysis. (default: parse-on-demand)
 ```
 
-### Taint analysis configuration <a name="taint"></a>
+##### Taint analysis configuration
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1535]:
```suggestion
#### Taint analysis configuration
```

File: docs/analyzer/user_guide.md [hunk 18]
```diff
@@ -1519,7 +1570,7 @@ The default configuration options of taint analysis are documented in the
 Clang SA's conceptual model of taint analysis and the checker's configuration
 file format is documented in the [Taint Analysis Configuration docs](https://clang.llvm.org/docs/analyzer/user-docs/TaintAnalysisConfiguration.html).
 
-### Statistical analysis mode <a name="statistical"></a>
+##### Statistical analysis mode
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:1573]:
```suggestion
#### Statistical analysis mode
```

File: docs/analyzer/user_guide.md [hunk 19]
```diff
@@ -1,56 +1,77 @@
...
+**static code analysis** and to store reports into a **web-based storage**.
+
+This document describes the **analysis** related sub-commands.
+
+The analysis related use-cases can be fully performend withhout a web-server.
+
+* [Integration of the analysis with your build system](#log)
+* [Execuing Static Analysis](#analyze)
+* [Showing analysis results in the command line](#parse)
+* [Applying code fixes](#fixes)
```
Comment (bruntib) [inline docs/analyzer/user_guide.md:64]:
```suggestion
* [Applying code fixes](#fixit)
```

**Review by bruntib — CHANGES_REQUESTED** [review]
Some typos could also be fixed. Please grep them, because some of these are occurring in the source code too (i.e. help pages).

performend
withhout
Execuing
analyse
analysers
expaneded
analysing
interations

**Comment by dkrupp** [comment]
Thanks for the review. I fixed everything.

**Review by bruntib — APPROVED** [review]

**Review by Szelethus — COMMENTED** [review]
I wonder how much the "JSON Compilation Commands" file is common knowledge. Maybe it could be a hyperlink to this: https://clang.llvm.org/docs/JSONCompilationDatabase.html. Also, while CodeChecker log is mentioned, there is no "More on compilation databases and CodeChecker log in section x.x". A reference to that section might be nice. It says to "see the individual help for log, analyze and parse (below in this User guide) for information about the arguments which are not documented here", but explicitly stating that all log basically does is that json database might be nice.

I was missing this under the "log" section as well. It states:
> The first step in performing an analysis on your project is to record information about the files in your project for the analyzers. This is done by recording a build of your project, which is done by the command CodeChecker log.

Why doesn't it state explicitly that what is being created is this once-mentioned JSON compilation database?

This is all notpicking, I couldn't get much further due to other issues I needed to attend to. Make a judgement call whether you want to address them.

**Comment by dkrupp** [comment]
> I wonder how much the "JSON Compilation Commands" file is common knowledge. Maybe it could be a hyperlink to this: https://clang.llvm.org/docs/JSONCompilationDatabase.html. Also, while CodeChecker log is mentioned, there is no "More on compilation databases and CodeChecker log in section x.x". A reference to that section might be nice. It says to "see the individual help for log, analyze and parse (below in this User guide) for information about the arguments which are not documented here", but explicitly stating that all log basically does is that json database might be nice.
> 
> I was missing this under the "log" section as well. It states:
> 
> > The first step in performing an analysis on your project is to record information about the files in your project for the analyzers. This is done by recording a build of your project, which is done by the command CodeChecker log.
> 
> Why doesn't it state explicitly that what is being created is this once-mentioned JSON compilation database?
> 
> This is all notpicking, I couldn't get much further due to other issues I needed to attend to. Make a judgement call whether you want to address them.

Thanks for the review. I extended the descriptions.

**Pull request merged**